### PR TITLE
chore(iast): native module safety improvements

### PR DESCRIPTION
**Pull request by juanjux**
## Description

In C, you can't import a module if there is an error set, so we can't sent metrics of an error (because we need to import the `_set_iast_error_metrics` symbol and we can't import it since there is an error...).

This will add a new `safe_import(module_name, symbol_name)` to `GenericUtils` that will store the error, import the module and symbol and restore the error (if there was any).

Also:

- Fix an issue with `has_pyerr_as_string()` where it was clearing the error.
- At an `atexit` callback to reset (and thus call the `reset_context`) to ensure it finishes before the Python environment.
- Small refactor of the the unittests since now they need the new import mechanism and the `Initializer` to be created.
- A couple small microfixes.

## Checklist
- [X] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp [hunk 1]
```diff
@@ -206,13 +264,15 @@ Initializer::create_context()
...
     ThreadContextCache.tx_map = map_ptr;
+    import_symbols_into_cache();
 }
 
 void
 Initializer::reset_context()
 {
     clear_tainting_maps();
     ThreadContextCache.tx_map = nullptr;
+    imported_cache.clear();
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp:275]: Do we really want to clear the cache on context reset?

Comment (juanjux) [reply ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp:275]: Yup. Segmentation fault if we dont :)

Comment (gnufede) [reply ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp:275]: Can't we call it only in `~Initializer()`? Don't we call this in between requests?

File: ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #include "Initializer.h"
 
+#include <iostream>
+#include <map>
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp:4]: We don't need this anymore right?
```suggestion
```

**Review by gnufede — APPROVED** [review]

File: ddtrace/appsec/_iast/_taint_tracking/Utils/GenericUtils.h [hunk 3]
```diff
@@ -24,17 +24,14 @@ is_iast_debug_enabled()
...
 
-inline py::object
-get_python_logger()
-{
-    py::object logger_module = py::module::import("ddtrace.internal.logger");
-    py::object get_logger = logger_module.attr("get_logger");
-    py::object native_logger = get_logger("native");
-    return native_logger;
-}
+inline pybind11::object
```
Comment (avara1986) [inline ddtrace/appsec/_iast/_taint_tracking/Utils/GenericUtils.h:27]:
```suggestion
inline py::object
```

File: ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp (hunk 2 above)
Comment (juanjux) [reply ddtrace/appsec/_iast/_taint_tracking/Initializer/Initializer.cpp:4]: Right.

**Pull request merged**